lessons/02-javascript/README.md

```diff
@@ -152,6 +157,8 @@ for (var key in object) {
 }
 ```
 
+Note that the order of iteration is not guarenteed with this syntax and later, if you use libraries which modify the prototype (more on this later) of base types, can have strange behavior.
```

Comment:
Yes, and I'm realizing now that I completely misread what the above code was trying to accomplish.

In my head I was advocating for `if ( p.hasOwnProperty(q) )` over `if ( p.q )`, which as a rule is a best practice for verifying if an object has a key. Somehow managed for form an argument around search paths without even thinking about it...